Chore: Bump storybook to v6 (#28926)
grafana/grafana: contribute/style-guides/storybook.md

}
```

### Controls

> The [controls addon](https://storybook.js.org/docs/react/essentials/controls) provides a way to interact with a component's properties dynamically and requires much less code than knobs. As such we're deprecating knobs in favour of using controls.

Reviewer: ```suggestion
The [controls addon](https://storybook.js.org/docs/react/essentials/controls) provides a way to interact with a component's properties dynamically and requires much less code than knobs. We're deprecating knobs in favor of using controls.
```

#### Migrating a story from Knobs to Controls

> The [button story](https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Button/Button.story.tsx) has been migrated as a test. Here's a basic guide on how to migrate a story to controls.

Reviewer: ```suggestion
As a test, we migrated the [button story](https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/components/Button/Button.story.tsx). Here's the guide on how to migrate a story to controls.
```

1.  Remove the `@storybook/addon-knobs` dependency.
2.  Import the Story type from `@storybook/react`

    `import { Story } from @storybook/react`

3.  Import the props interface from the component you're working on (these must be exported in the component).

    `import { Props } from './Component'`

> 4.  Add the Story type to all stories in the file, replace the props sent to the component

Reviewer: ```suggestion
4.  Add the Story type to all stories in the file, then replace the props sent to the component
```

    and remove any knobs.

    Before

    ```tsx
    export const Simple = () => {
      const prop1 = text('Prop1', 'Example text');
      const prop2 = select('Prop2', ['option1', 'option2'], 'option1');

      return <Component prop1={prop1} prop2={prop2} />;
    };
    ```

    After

    ```tsx
    export const Simple: Story<Props> = ({ prop1, prop2 }) => {
      return <Component prop1={prop1} prop2={prop2} />;
    };
    ```

5.  Add default props (or args in Storybook language).

    ```tsx
    Simple.args = {
      prop1: 'Example text',
      prop2: 'option 1',
    };
    ```

> 6.  If the component has advanced props type (ie. other than string, number, boolean). You need to

Reviewer: ```suggestion
6.  If the component has advanced props type (ie. other than string, number, boolean), you need to
```
